### PR TITLE
LPS-78821 Implement UAD for BlogsEntry

### PR DESCRIPTION
**Pull request by willnewbury**
/cc @drewbrokke @noahsherrillwork

### CODE REVIEW

**Comment by willnewbury** [comment]
ci:test

File: modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+@Component(immediate = true, service = BlogsEntryUADEntityTestHelper.class)
+public class BlogsEntryUADEntityTestHelper {
+
+	public BlogsEntry addBlogsEntry(long userId) throws Exception {
+		Calendar calendar = CalendarFactoryUtil.getCalendar();
+
+		calendar.add(Calendar.DATE, 1);
+
+		Date displayDate = calendar.getTime();
```
Comment (noahsherrillwork) [inline modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java]: `displayDate` can be eliminated by passing `calendar.getTime()` directly to the `addEntry()` method.

File: modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brian Wing Shun Chan
```
Comment (noahsherrillwork) [inline modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java]: The `@author` annotation should be changed to your name, as this is a manual implementation.

**Comment by willnewbury** [comment]
ci:reopen

File: modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		return _blogsEntryLocalService.addEntry(
+			userId, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), displayDate, serviceContext);
+	}
+
+	public BlogsEntry addBlogsEntryWithStatusByUserId(
+			long userId, long statusByUserId)
+		throws Exception {
+
+		BlogsEntry bookmarksEntry = addBlogsEntry(userId);
```
Comment (noahsherrillwork) [inline modules/apps/collaboration/blogs/blogs-uad-test/src/testIntegration/java/com/liferay/blogs/uad/test/BlogsEntryUADEntityTestHelper.java]: `bookmarksEntry` should be `blogsEntry`.

File: modules/apps/collaboration/blogs/blogs-uad/src/main/java/com/liferay/blogs/uad/display/BlogsEntryUADEntityDisplayHelper.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import com.liferay.portal.kernel.util.Portal;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletURL;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brian Wing Shun Chan
```
Comment (noahsherrillwork) [inline modules/apps/collaboration/blogs/blogs-uad/src/main/java/com/liferay/blogs/uad/display/BlogsEntryUADEntityDisplayHelper.java]: The `@author` annotation should be changed to your name, as this is a manual implementation.

File: modules/apps/collaboration/blogs/blogs-uad/src/main/java/com/liferay/blogs/uad/display/BlogsEntryUADEntityDisplayHelper.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+		String portletId = PortletProviderUtil.getPortletId(
+			BlogsEntry.class.getName(), PortletProvider.Action.VIEW);
+
+		PortletURL portletURL = liferayPortletResponse.createLiferayPortletURL(
+			portal.getControlPanelPlid(liferayPortletRequest), portletId,
+			PortletRequest.RENDER_PHASE);
+
+		portletURL.setParameter("mvcRenderCommandName", "/blogs/edit_entry");
+		portletURL.setParameter(
```
Comment (noahsherrillwork) [inline modules/apps/collaboration/blogs/blogs-uad/src/main/java/com/liferay/blogs/uad/display/BlogsEntryUADEntityDisplayHelper.java]: I think this is supposed to follow the same order as in jsp files, i.e. `mvcRenderCommandName` then `redirect` then other parameters.

**Comment by willnewbury** [comment]
I'll resend with changes

**Pull request closed without merging**